app/assets/javascripts/sitewide/drag_n_drop.js.erb

```diff
@@ -181,31 +210,51 @@
       return fd;
     }
 
+    function _disableSubmitButton() {
+      $('.save-result').prop('disabled', true);
+    }
+
+    function _enableSubmitButton() {
+      $('.save-result').prop('disabled', false);
+    }
+
     function _filerAndCheckFiles() {
-      droppedFiles = droppedFiles.filter(Boolean);
+      for(var i = 0; i < droppedFiles.length; i++) {
+        if(droppedFiles[i].isValid == false) {
+          return false;
+        }
+      }
       return (droppedFiles.length > 0);
     }
 
     function _validateFilesSize(file) {
-      var maxSize = file.size/1048576;
-      if(maxSize > <%= Constants::FILE_MAX_SIZE_MB %> && isValid) {
-        return "<p><%= I18n.t 'general.file.size_exceeded', file_size: Constants::FILE_MAX_SIZE_MB %></p>";
+      var fileSize = file.size;
+      totalSize += parseInt(fileSize);
+      if(fileSize > <%= Constants::FILE_MAX_SIZE_MB.megabyte %>) {
```

Comment:
we may convert the FILE_MAX_SIZE_MB in the Constants file or rename to FILE_MAX_SIZE because one can think that this is already convert in MB

Reply:
But this constant contains MB indeed, I just convert it to bytes for more precise calculations.
It also used in messages, so why rename it?

Reply:
so why don't rather convert fileSize in MB

Reply:
For more accurate calculations in bytes

Reply:
yes but is still not so accurate I uploaded 10 files with 5,2 MB and still let me pass so  ..

Reply:
What is size of that file in bytes?

Reply:
5166441